File: Advanced Programming Methods/Laboratoare/Lab/Lab13/Lab12/Lab12/validator/TemaValidator.cs
using System;
using System.Collections.Generic;
using System.Text;
using Lab12.domain;

namespace Lab12.validator
{
    class TemaValidator:IValidator<Tema>
    {
        public TemaValidator() { }
        public void Validate(Tema t)
        {
            String err = "";
            if (t.Id < 0 || t.Id > 14)
                err += "Numar tema invalid!\n";
            if (t.Deadline < 1 || t.Deadline > 14)
                err += "Termen limita de predare invalid!\n";
            if (t.SaptPrimire < 1 || t.SaptPrimire > 14)
                err += "Saptamana deprimire tema invalida!\n";
            if (t.SaptPrimire > t.Deadline)
                err += "Saptamana cu deadline-ul trebuie sa fie aceeasi sau sa urmeza saptamanii de primire!\n";
            if (err != "")
                throw new ValidationException(err);

        }
    }
}

File: Advanced Programming Methods/Laboratoare/Lab/Lab3,4/src/main/java/factory/TemaValidator.java
package factory;

import domain.Tema;

/**
 * TemaValidator implements Validator interface - E=Tema
 */
public class TemaValidator implements Validator<Tema> {

    /**
     * validates a student
     * @param t - Tema object to be validated
     * @throws ValidationException if the id, deadline or week of receiving of the Tema object are not valid
     */
    public void validate(Tema t){
        String err="";
        if(t.getID()<0 || t.getID()>14)
            err+="Numar tema invalid!\n";
        if(t.getDeadline()<1 || t.getDeadline()>14)
            err+="Termen limita de predare invalid!\n";
        if(t.getSaptPrimireTema()<1 || t.getSaptPrimireTema()>14)
            err+="Saptamana deprimire tema invalida!\n";
        if(err!="")
            throw new ValidationException(err);
    }
}
